tests/integration_tests/bugs/test_lp1897099.py

```diff
@@ -0,0 +1,39 @@
+""" Integration test for LP #187099
+
+Ensure that if fallocate fails during mkswap that we fall back to using dd
+
+https://bugs.launchpad.net/cloud-init/+bug/1897099
+"""
+
+import pytest
+
+
+USER_DATA = """\
+#cloud-config
+swap:
+  filename: /swap.img
+  size: 10000000
+  maxsize: 10000000
+"""
+
+
+@pytest.mark.user_data(USER_DATA)
+@pytest.mark.lxd_vm
+def test_fallocate_fallback(client):
+    # Setup instance
+    client.execute('swapoff -a')
+    client.execute('rm -r /swap.img')
+    client.execute("echo 'whoops' > /usr/bin/fallocate")
+
+    # Reset instance state
+    client.execute('cloud-init clean --logs && sync')
+    client.instance.restart()
+    client.instance.wait(raise_on_cloudinit_failure=False)
```

Comment:
Did you find you needed this? I didn't in https://github.com/canonical/cloud-init/blob/66b4be8b6da188a0667bd8c86a25155b6f4f3f6c/tests/integration_tests/bugs/test_lp1900837.py#L25 because [`.restart()`](https://github.com/canonical/pycloudlib/blob/master/pycloudlib/lxd/instance.py#L215) _should_ wait by default:

```py
     def restart(self, wait=True, force=False, **kwargs):
```

My test has only run in a container, AFAIK, so it wouldn't surprise me if there were some issue with VMs.

(Thinking a little more, I guess the fact that `restart`s wait will raise on cloud-init failure may be a more general issue?)

Reply:
I didn't try without the `restart`, but with the bootcmd it doesn't matter anyway. And yes, raising on cloud-init failure may turn out to be a problem.